mm2src/coins/z_coin.rs

```diff
@@ -1387,7 +1428,8 @@ impl SwapOps for ZCoin {
                 )))
             },
         };
-        let amount_sat = validate_fee_args.dex_fee.fee_uamount(self.utxo_arc.decimals)?;
+        let fee_amount_sat = validate_fee_args.dex_fee.fee_amount_as_u64(self.utxo_arc.decimals)?;
+        let burn_amount_sat = validate_fee_args.dex_fee.burn_amount_as_u64(self.utxo_arc.decimals)?;
         let expected_memo = MemoBytes::from_bytes(validate_fee_args.uuid).expect("Uuid length < 512");
```

Comment:
Can we check if `z_tx.shielded_outputs.is_empty()` and return early error before performing these operations.

Reply:
Do we really need this?
If there is no shielded_outputs or less than expected, we would have one of fee_output_valid or burn_output_valid false. Looks like an extra check for me